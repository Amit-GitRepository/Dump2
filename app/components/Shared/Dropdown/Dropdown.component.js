import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Dropdown.styles.js';
import {Button, Card, Icon, ISText, List, Modal, Touchable} from '../../Shared';
import {colors, fonts} from '../../../themes/constants.styles';
import {translate} from '../../../language/i18n/helper';
import {View} from 'react-native';

class Dropdown extends Component {
  state = {
    isVisible: false
  };

  showModal = () => this.setState({isVisible: true});
  hideModal = () => this.setState({isVisible: false});

  selectItem = (item) => () => {
    if (item !== this.props.value) {
      this.props.onSelect(item); // don't call onSelect if the value is same
    }
    this.hideModal();
  }
  render () {
    const {accessibilityLabel, data, unselectedText, value, style, testID} = this.props;
    const modalHeaderText = this.props.headerText || this.props.unselectedText;
    return (
      <View style={styles.container} accessibilityLabel={accessibilityLabel} testID={testID}>
        <Touchable onPress={this.showModal} style={{...styles.touchableContainer, ...style}} >
          <View style={styles.viewContainer}>
            <ISText style= {styles.title} type={'BOLD'}>{value || unselectedText}</ISText>
            <Icon name={'caret-down'} size={fonts.FONT_SIZE_XL} color={colors.PRIMARY_ACTIONABLE} style={styles.downArrow}/>
          </View>
        </Touchable>

        <Modal isVisible={this.state.isVisible} animationIn={'zoomIn'} animationOut={'zoomOut'}
          animationInTiming={1} backdropTransitionInTiming={1} backdropTransitionOutTiming={1}
          animationOutTiming={1} onBackdropPress={this.hideModal}>
          <Card header={modalHeaderText} >
            <View style={styles.listContainer}>
              <List data={data} alignListItemsCenter onPress={this.selectItem} arrowVisibility={false}/>
            </View>
          </Card>
          <Button onPress={this.hideModal} text={translate('STORE_LOCATOR__CANCEL')} style={styles.cancelButton}/>
        </Modal>
      </View>
    );
  }
}

Dropdown.defaultProps = {
  data: [],
  unselectedText: translate('DROPDOWN_PLEASE_SELECT'),
  headerText: '',
  onSelect: noop
};
Dropdown.propTypes = {
  accessibilityLabel: PropTypes.string,
  data: PropTypes.array,
  unselectedText: PropTypes.string,
  headerText: PropTypes.string,
  onSelect: PropTypes.func,
  value: PropTypes.string,
  style: PropTypes.object,
  testID: PropTypes.string
};
export default Dropdown;
