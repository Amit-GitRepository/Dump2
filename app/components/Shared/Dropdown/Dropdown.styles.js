import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    flex: 1
  },
  touchableContainer: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    paddingHorizontal: 10,
    paddingVertical: 6,
    borderRadius: 8,
    borderColor: colors.PRIMARY_ACTIONABLE,
    borderWidth: 1
  },
  viewContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  listContainer: {
    marginTop: -15,
    borderRadius: 10,
    paddingBottom: 75
  },
  cancelButton: {
    marginTop: 10
  },
  title: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  downArrow: {
    marginTop: 3
  }
};
