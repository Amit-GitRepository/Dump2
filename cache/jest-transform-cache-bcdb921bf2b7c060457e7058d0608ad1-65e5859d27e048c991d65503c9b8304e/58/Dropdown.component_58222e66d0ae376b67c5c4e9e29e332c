Object.defineProperty(exports,"__esModule",{value:true});var cov_mo5m00yqv=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Dropdown/Dropdown.component.js',hash='1aa3b4bc78d2e47a5528ea017b9749d186d5158c',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Dropdown/Dropdown.component.js',statementMap:{'0':{start:{line:20,column:20},end:{line:20,column:52}},'1':{start:{line:21,column:20},end:{line:21,column:53}},'2':{start:{line:23,column:25},end:{line:28,column:3}},'3':{start:{line:24,column:4},end:{line:26,column:5}},'4':{start:{line:25,column:6},end:{line:25,column:32}},'5':{start:{line:27,column:4},end:{line:27,column:21}},'6':{start:{line:30,column:77},end:{line:30,column:87}},'7':{start:{line:31,column:28},end:{line:31,column:78}},'8':{start:{line:32,column:4},end:{line:52,column:6}},'9':{start:{line:56,column:0},end:{line:60,column:2}},'10':{start:{line:61,column:0},end:{line:70,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:20,column:14},end:{line:20,column:15}},loc:{start:{line:20,column:20},end:{line:20,column:52}},line:20},'1':{name:'(anonymous_1)',decl:{start:{line:21,column:14},end:{line:21,column:15}},loc:{start:{line:21,column:20},end:{line:21,column:53}},line:21},'2':{name:'(anonymous_2)',decl:{start:{line:23,column:15},end:{line:23,column:16}},loc:{start:{line:23,column:25},end:{line:28,column:3}},line:23},'3':{name:'(anonymous_3)',decl:{start:{line:23,column:25},end:{line:23,column:26}},loc:{start:{line:23,column:31},end:{line:28,column:3}},line:23},'4':{name:'(anonymous_4)',decl:{start:{line:29,column:2},end:{line:29,column:3}},loc:{start:{line:29,column:12},end:{line:53,column:3}},line:29}},branchMap:{'0':{loc:{start:{line:24,column:4},end:{line:26,column:5}},type:'if',locations:[{start:{line:24,column:4},end:{line:26,column:5}},{start:{line:24,column:4},end:{line:26,column:5}}],line:24},'1':{loc:{start:{line:31,column:28},end:{line:31,column:78}},type:'binary-expr',locations:[{start:{line:31,column:28},end:{line:31,column:49}},{start:{line:31,column:53},end:{line:31,column:78}}],line:31},'2':{loc:{start:{line:36,column:57},end:{line:36,column:80}},type:'binary-expr',locations:[{start:{line:36,column:57},end:{line:36,column:62}},{start:{line:36,column:66},end:{line:36,column:80}}],line:36}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Button=require('../Button/Button.component');var _Button2=_interopRequireDefault(_Button);
var _Card=require('../Card/Card.component');var _Card2=_interopRequireDefault(_Card);
var _Icon=require('../Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _List=require('../List/List.component');var _List2=_interopRequireDefault(_List);
var _Modal=require('../Modal/Modal.component');var _Modal2=_interopRequireDefault(_Modal);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _DropdownStyles=require('./Dropdown.styles.js');var _DropdownStyles2=_interopRequireDefault(_DropdownStyles);
var _Touchable=require('../Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _constants=require('../../../themes/constants.styles');
var _Shared=require('../../Shared');
var _helper=require('../../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

Dropdown=function(_Component){_inherits(Dropdown,_Component);function Dropdown(){var _ref;var _temp,_this,_ret;_classCallCheck(this,Dropdown);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=Dropdown.__proto__||Object.getPrototypeOf(Dropdown)).call.apply(_ref,[this].concat(args))),_this),_this.
state={
isVisible:false},_this.


showModal=function(){cov_mo5m00yqv.f[0]++;cov_mo5m00yqv.s[0]++;return _this.setState({isVisible:true});},_this.
hideModal=function(){cov_mo5m00yqv.f[1]++;cov_mo5m00yqv.s[1]++;return _this.setState({isVisible:false});},_this.

selectItem=function(item){cov_mo5m00yqv.f[2]++;cov_mo5m00yqv.s[2]++;return function(){cov_mo5m00yqv.f[3]++;cov_mo5m00yqv.s[3]++;
if(item!==_this.props.value){cov_mo5m00yqv.b[0][0]++;cov_mo5m00yqv.s[4]++;
_this.props.onSelect(item);
}else{cov_mo5m00yqv.b[0][1]++;}cov_mo5m00yqv.s[5]++;
_this.hideModal();
};},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(Dropdown,[{key:'render',value:function render()
{cov_mo5m00yqv.f[4]++;var _ref2=(cov_mo5m00yqv.s[6]++,
this.props),accessibilityLabel=_ref2.accessibilityLabel,data=_ref2.data,unselectedText=_ref2.unselectedText,value=_ref2.value,style=_ref2.style,testID=_ref2.testID;
var modalHeaderText=(cov_mo5m00yqv.s[7]++,(cov_mo5m00yqv.b[1][0]++,this.props.headerText)||(cov_mo5m00yqv.b[1][1]++,this.props.unselectedText));cov_mo5m00yqv.s[8]++;
return(
_react2.default.createElement(_reactNative.View,{style:_DropdownStyles2.default.container,accessibilityLabel:accessibilityLabel,testID:testID},
_react2.default.createElement(_Touchable2.default,{onPress:this.showModal,style:_extends({},_DropdownStyles2.default.touchableContainer,style)},
_react2.default.createElement(_reactNative.View,{style:_DropdownStyles2.default.viewContainer},
_react2.default.createElement(_Shared.ISText,{style:_DropdownStyles2.default.title,type:'BOLD'},(cov_mo5m00yqv.b[2][0]++,value)||(cov_mo5m00yqv.b[2][1]++,unselectedText)),
_react2.default.createElement(_Icon2.default,{name:'caret-down',size:_constants.fonts.FONT_SIZE_XL,color:_constants.colors.PRIMARY_ACTIONABLE,style:_DropdownStyles2.default.downArrow}))),



_react2.default.createElement(_Modal2.default,{isVisible:this.state.isVisible,animationIn:'zoomIn',animationOut:'zoomOut',
animationInTiming:1,backdropTransitionInTiming:1,backdropTransitionOutTiming:1,
animationOutTiming:1,onBackdropPress:this.hideModal},
_react2.default.createElement(_Card2.default,{header:modalHeaderText},
_react2.default.createElement(_reactNative.View,{style:_DropdownStyles2.default.listContainer},
_react2.default.createElement(_List2.default,{data:data,alignListItemsCenter:true,onPress:this.selectItem,arrowVisibility:false}))),


_react2.default.createElement(_Button2.default,{onPress:this.hideModal,text:(0,_helper.translate)('STORE_LOCATOR__CANCEL'),style:_DropdownStyles2.default.cancelButton}))));



}}]);return Dropdown;}(_react.Component);cov_mo5m00yqv.s[9]++;


Dropdown.defaultProps={
data:[],
unselectedText:(0,_helper.translate)('DROPDOWN_PLEASE_SELECT'),
headerText:''};cov_mo5m00yqv.s[10]++;

Dropdown.propTypes={
accessibilityLabel:_propTypes2.default.string,
data:_propTypes2.default.array,
unselectedText:_propTypes2.default.string,
headerText:_propTypes2.default.string,
onSelect:_propTypes2.default.func,
value:_propTypes2.default.string,
style:_propTypes2.default.object,
testID:_propTypes2.default.string};exports.default=

Dropdown;