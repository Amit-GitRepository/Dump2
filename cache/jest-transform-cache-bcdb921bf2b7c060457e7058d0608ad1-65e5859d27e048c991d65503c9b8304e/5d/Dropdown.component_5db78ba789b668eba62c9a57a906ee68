Object.defineProperty(exports,"__esModule",{value:true});var cov_mo5m00yqv=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Dropdown/Dropdown.component.js',hash='b749e51ffdbf53a1a1eb6cd0b8ad04789f14f684',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Dropdown/Dropdown.component.js',statementMap:{'0':{start:{line:15,column:20},end:{line:15,column:52}},'1':{start:{line:16,column:20},end:{line:16,column:53}},'2':{start:{line:18,column:25},end:{line:23,column:3}},'3':{start:{line:19,column:4},end:{line:21,column:5}},'4':{start:{line:20,column:6},end:{line:20,column:32}},'5':{start:{line:22,column:4},end:{line:22,column:21}},'6':{start:{line:25,column:77},end:{line:25,column:87}},'7':{start:{line:26,column:28},end:{line:26,column:78}},'8':{start:{line:27,column:4},end:{line:47,column:6}},'9':{start:{line:51,column:0},end:{line:56,column:2}},'10':{start:{line:57,column:0},end:{line:66,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:15,column:14},end:{line:15,column:15}},loc:{start:{line:15,column:20},end:{line:15,column:52}},line:15},'1':{name:'(anonymous_1)',decl:{start:{line:16,column:14},end:{line:16,column:15}},loc:{start:{line:16,column:20},end:{line:16,column:53}},line:16},'2':{name:'(anonymous_2)',decl:{start:{line:18,column:15},end:{line:18,column:16}},loc:{start:{line:18,column:25},end:{line:23,column:3}},line:18},'3':{name:'(anonymous_3)',decl:{start:{line:18,column:25},end:{line:18,column:26}},loc:{start:{line:18,column:31},end:{line:23,column:3}},line:18},'4':{name:'(anonymous_4)',decl:{start:{line:24,column:2},end:{line:24,column:3}},loc:{start:{line:24,column:12},end:{line:48,column:3}},line:24}},branchMap:{'0':{loc:{start:{line:19,column:4},end:{line:21,column:5}},type:'if',locations:[{start:{line:19,column:4},end:{line:21,column:5}},{start:{line:19,column:4},end:{line:21,column:5}}],line:19},'1':{loc:{start:{line:26,column:28},end:{line:26,column:78}},type:'binary-expr',locations:[{start:{line:26,column:28},end:{line:26,column:49}},{start:{line:26,column:53},end:{line:26,column:78}}],line:26},'2':{loc:{start:{line:31,column:57},end:{line:31,column:80}},type:'binary-expr',locations:[{start:{line:31,column:57},end:{line:31,column:62}},{start:{line:31,column:66},end:{line:31,column:80}}],line:31}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _DropdownStyles=require('./Dropdown.styles.js');var _DropdownStyles2=_interopRequireDefault(_DropdownStyles);
var _Shared=require('../../Shared');
var _constants=require('../../../themes/constants.styles');
var _helper=require('../../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

Dropdown=function(_Component){_inherits(Dropdown,_Component);function Dropdown(){var _ref;var _temp,_this,_ret;_classCallCheck(this,Dropdown);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=Dropdown.__proto__||Object.getPrototypeOf(Dropdown)).call.apply(_ref,[this].concat(args))),_this),_this.
state={
isVisible:false},_this.


showModal=function(){cov_mo5m00yqv.f[0]++;cov_mo5m00yqv.s[0]++;return _this.setState({isVisible:true});},_this.
hideModal=function(){cov_mo5m00yqv.f[1]++;cov_mo5m00yqv.s[1]++;return _this.setState({isVisible:false});},_this.

selectItem=function(item){cov_mo5m00yqv.f[2]++;cov_mo5m00yqv.s[2]++;return function(){cov_mo5m00yqv.f[3]++;cov_mo5m00yqv.s[3]++;
if(item!==_this.props.value){cov_mo5m00yqv.b[0][0]++;cov_mo5m00yqv.s[4]++;
_this.props.onSelect(item);
}else{cov_mo5m00yqv.b[0][1]++;}cov_mo5m00yqv.s[5]++;
_this.hideModal();
};},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(Dropdown,[{key:'render',value:function render()
{cov_mo5m00yqv.f[4]++;var _ref2=(cov_mo5m00yqv.s[6]++,
this.props),accessibilityLabel=_ref2.accessibilityLabel,data=_ref2.data,unselectedText=_ref2.unselectedText,value=_ref2.value,style=_ref2.style,testID=_ref2.testID;
var modalHeaderText=(cov_mo5m00yqv.s[7]++,(cov_mo5m00yqv.b[1][0]++,this.props.headerText)||(cov_mo5m00yqv.b[1][1]++,this.props.unselectedText));cov_mo5m00yqv.s[8]++;
return(
_react2.default.createElement(_reactNative.View,{style:_DropdownStyles2.default.container,accessibilityLabel:accessibilityLabel,testID:testID},
_react2.default.createElement(_Shared.Touchable,{onPress:this.showModal,style:_extends({},_DropdownStyles2.default.touchableContainer,style)},
_react2.default.createElement(_reactNative.View,{style:_DropdownStyles2.default.viewContainer},
_react2.default.createElement(_Shared.ISText,{style:_DropdownStyles2.default.title,type:'BOLD'},(cov_mo5m00yqv.b[2][0]++,value)||(cov_mo5m00yqv.b[2][1]++,unselectedText)),
_react2.default.createElement(_Shared.Icon,{name:'caret-down',size:_constants.fonts.FONT_SIZE_XL,color:_constants.colors.PRIMARY_ACTIONABLE,style:_DropdownStyles2.default.downArrow}))),



_react2.default.createElement(_Shared.Modal,{isVisible:this.state.isVisible,animationIn:'zoomIn',animationOut:'zoomOut',
animationInTiming:1,backdropTransitionInTiming:1,backdropTransitionOutTiming:1,
animationOutTiming:1,onBackdropPress:this.hideModal},
_react2.default.createElement(_Shared.Card,{header:modalHeaderText},
_react2.default.createElement(_reactNative.View,{style:_DropdownStyles2.default.listContainer},
_react2.default.createElement(_Shared.List,{data:data,alignListItemsCenter:true,onPress:this.selectItem,arrowVisibility:false}))),


_react2.default.createElement(_Shared.Button,{onPress:this.hideModal,text:(0,_helper.translate)('STORE_LOCATOR__CANCEL'),style:_DropdownStyles2.default.cancelButton}))));



}}]);return Dropdown;}(_react.Component);cov_mo5m00yqv.s[9]++;


Dropdown.defaultProps={
data:[],
unselectedText:(0,_helper.translate)('DROPDOWN_PLEASE_SELECT'),
headerText:'',
onSelect:_noop2.default};cov_mo5m00yqv.s[10]++;

Dropdown.propTypes={
accessibilityLabel:_propTypes2.default.string,
data:_propTypes2.default.array,
unselectedText:_propTypes2.default.string,
headerText:_propTypes2.default.string,
onSelect:_propTypes2.default.func,
value:_propTypes2.default.string,
style:_propTypes2.default.object,
testID:_propTypes2.default.string};exports.default=

Dropdown;