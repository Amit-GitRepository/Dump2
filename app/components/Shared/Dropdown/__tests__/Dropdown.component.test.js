import Dropdown from '../Dropdown.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('Dropdown component', () => {
  it('should render correctly', () => {
    const componentTree = renderer.create(<Dropdown />).toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  it('should set isVisible in state to true if showModal is called', () => {
    const wrapper = shallow(<Dropdown/>);
    wrapper.setState({isVisible: false});
    wrapper.instance().showModal();
    expect(wrapper.instance().state).toEqual({isVisible: true});
  });

  it('should set isVisible in state to false', () => {
    const wrapper = shallow(<Dropdown/>);
    wrapper.setState({isVisible: true});
    wrapper.instance().hideModal();
    expect(wrapper.instance().state).toEqual({isVisible: false});
  });

  it('should set isVisible in state to false when selectItem is called', () => {
    const wrapper = shallow(<Dropdown onSelect={jest.fn()}/>);
    wrapper.setState({isVisible: true});
    wrapper.instance().selectItem('test')();
    expect(wrapper.instance().state).toEqual({isVisible: false});
  });

  it('should set isVisible in state to false when selectItem is called', () => {
    const onSelect = jest.fn((val) => val);
    const wrapper = shallow(<Dropdown onSelect={onSelect}/>);
    wrapper.setState({isVisible: true});
    wrapper.instance().selectItem('test')();
    expect(onSelect).toHaveBeenCalledWith('test');
  });
});
