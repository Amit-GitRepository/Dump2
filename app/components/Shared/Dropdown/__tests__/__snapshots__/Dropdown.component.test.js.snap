// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render correctly 1`] = `
<View
  accessibilityLabel={undefined}
  style={
    Object {
      "flex": 1,
    }
  }
  testID={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#F93E60",
        "borderRadius": 8,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 6,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#303C46",
              "fontFamily": "SukhumvitSet-Bold",
              "fontSize": 18,
            }
          }
        >
          Please select
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#F93E60",
                "fontSize": 24,
              },
              Object {
                "marginTop": 3,
              },
              Object {
                "fontFamily": "iservice",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(48,60,70,0.9)",
            "height": 1334,
            "width": 750,
          },
        ]
      }
      testID={undefined}
    />
    <View
      onModalHide={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        elevation={5}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flex": 1,
              "marginHorizontal": -1,
              "overflow": "hidden",
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {},
          ]
        }
      >
        <View
          onPress={null}
          style={
            Array [
              Object {
                "zIndex": 1,
              },
              Object {},
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "borderWidth": 0,
                  "elevation": 3,
                  "shadowColor": "#303C46",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#F93E60",
                      "paddingHorizontal": 5,
                    },
                    Object {
                      "borderBottomColor": "#303C46",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#E9E9EA",
                          "padding": 10,
                        },
                        Object {
                          "backgroundColor": "#303C46",
                          "marginHorizontal": -5,
                          "padding": 10,
                        },
                      ],
                      Object {},
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 18,
                          },
                          Object {
                            "color": "#FFFFFF",
                          },
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    Please select
                  </Text>
                </View>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#303C46",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "bottom": 9.6,
                      "marginBottom": -9.6,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 10,
              "marginTop": -15,
              "paddingBottom": 75,
            }
          }
        >
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#F93E60",
            "borderRadius": 50,
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 16,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-Bold",
                  },
                ]
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
